### PR TITLE
Add SanitizeTags to stable CreateFromSource

### PR DESCRIPTION
**Pull request by AKhoria**
## Change Overview

Add SanitizeTags to stable CreateFromSource
+ Unit test which checks that we are able to obtain a created snapshot through List with tags

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/snapshot/snapshot_test.go [hunk 1]
```diff
@@ -1015,6 +1015,49 @@ func (s *SnapshotTestSuite) TestCreateFromSourceBeta(c *C) {
...
+	snapshoterStable := snapshot.NewSnapshotStable(kubeCli, dynCli)
+
+	for _, snapshoter := range []snapshot.Snapshotter{snapshoterAlpha, snapshoterBeta, snapshoterStable} {
+		err := snapshoter.CreateFromSource(ctx, source, snapshotName, namespace, false, labels)
+		c.Assert(err, IsNil)
+		foundSns, err := snapshoter.List(ctx, namespace, labels)
+		c.Assert(err, IsNil)
+		c.Assert(foundSns.Items, HasLen, 1)
+		c.Assert(foundSns.Items[0].Name, Equals, snapshotName)
+	}
```
Comment (viveksinghggits) [inline pkg/kube/snapshot/snapshot_test.go:1058]: after creation can be list these to make sure we are able to list the created snapshots?

**Review by viveksinghggits — APPROVED** [review]

File: pkg/kube/snapshot/snapshot_stable.go [hunk 2]
```diff
@@ -126,7 +127,7 @@ func (sna *SnapshotStable) CreateFromSource(ctx context.Context, source *Source,
 		return errors.Wrap(err, "Failed to get DeletionPolicy from VolumeSnapshotClass")
 	}
 	contentName := snapshotName + "-content-" + string(uuid.NewUUID())
-	snap := UnstructuredVolumeSnapshot(VolSnapGVR, snapshotName, namespace, "", contentName, source.VolumeSnapshotClassName, labels)
+	snap := UnstructuredVolumeSnapshot(VolSnapGVR, snapshotName, namespace, "", contentName, source.VolumeSnapshotClassName, blockstorage.SanitizeTags(labels))
```
Comment (pavannd1) [inline pkg/kube/snapshot/snapshot_stable.go:130]:
```suggestion
	snap := UnstructuredVolumeSnapshot(
		VolSnapGVR,
		snapshotName,
		namespace,
		"",
		contentName,
		source.VolumeSnapshotClassName,
		blockstorage.SanitizeTags(labels),
	)
```
[nit] formatting

File: pkg/kube/snapshot/snapshot_test.go [hunk 3]
```diff
@@ -1015,6 +1015,49 @@ func (s *SnapshotTestSuite) TestCreateFromSourceBeta(c *C) {
...
+	scheme := runtime.NewScheme()
+
+	volSnap := snapshot.UnstructuredVolumeSnapshot(v1alpha1.VolSnapGVR, existingSnapshotName, namespace, "pvcName", "content", snapshotClass, nil)
+	volSnapClass := snapshot.UnstructuredVolumeSnapshotClass(v1alpha1.VolSnapClassGVR, snapshotClass, "driver", "DELETE", nil)
+	dynCli := dynfake.NewSimpleDynamicClient(scheme, volSnap, volSnapClass)
+	kubeCli := fake.NewSimpleClientset()
+	snapshoterAlpha := snapshot.NewSnapshotAlpha(kubeCli, dynCli)
+
+	volSnap = snapshot.UnstructuredVolumeSnapshot(v1beta1.VolSnapGVR, existingSnapshotName, namespace, "pvcName", "content", snapshotClass, nil)
```
Comment (pavannd1) [inline pkg/kube/snapshot/snapshot_test.go:1039]:
```suggestion
	volSnap = snapshot.UnstructuredVolumeSnapshot(
		v1beta1.VolSnapGVR,
		existingSnapshotName,
		namespace,
		"pvcName",
		"content",
		snapshotClass,
		nil,
	)
```

File: pkg/kube/snapshot/snapshot_test.go [hunk 4]
```diff
@@ -1015,6 +1015,49 @@ func (s *SnapshotTestSuite) TestCreateFromSourceBeta(c *C) {
...
+	volSnapClass = snapshot.UnstructuredVolumeSnapshotClass(v1beta1.VolSnapClassGVR, snapshotClass, "driver", "DELETE", nil)
+	dynCli = dynfake.NewSimpleDynamicClient(scheme, volSnap, volSnapClass)
+	kubeCli = fake.NewSimpleClientset()
+	snapshoterBeta := snapshot.NewSnapshotBeta(kubeCli, dynCli)
+
+	volSnap = snapshot.UnstructuredVolumeSnapshot(snapshot.VolSnapGVR, existingSnapshotName, namespace, "pvcName", "content", snapshotClass, nil)
```
Comment (pavannd1) [inline pkg/kube/snapshot/snapshot_test.go:1045]:
```suggestion
	volSnap = snapshot.UnstructuredVolumeSnapshot(
		snapshot.VolSnapGVR,
		existingSnapshotName,
		namespace,
		"pvcName",
		"content",
		snapshotClass,
		nil,
	)
```

**Pull request merged**